src/loki/search.cc

```diff
@@ -491,11 +491,11 @@ struct bin_handler_t {
 
       // if this edge is filtered
       const auto* edge = tile->directededge(edge_id);
-      if (!costing->Allowed(edge, tile)) {
+      if (edge->is_shortcut() || !costing->Allowed(edge, tile)) {
         // then we try its opposing edge
         edge_id = reader.GetOpposingEdgeId(edge_id, edge, tile);
         // but if we couldnt get it or its filtered too then we move on
-        if (!edge_id.Is_Valid() || !costing->Allowed(edge, tile))
+        if (edge->is_shortcut() || !edge_id.Is_Valid() || !costing->Allowed(edge, tile))
```

Comment:
edge will be null if the id is invalid so we have to check the id first
```suggestion
        if (!edge_id.Is_Valid() || edge->is_shortcut() || !costing->Allowed(edge, tile))
```